telegram/ext/_updater.py

```diff
@@ -17,174 +17,146 @@
+        * The only arguments and attributes are now :attr:`bot` and :attr:`update_queue` as now
+          the sole purpose of this class is to fetch updates. The entry point to a PTB application
+          is now :class:`telegram.ext.Application`.
 
     Attributes:
         bot (:class:`telegram.Bot`): The bot used with this Updater.
-        user_signal_handler (Callable): Optional. Function to be called when a signal is
-            received.
-
-            .. versionchanged:: 14.0
-                Renamed ``user_sig_handler`` to ``user_signal_handler``.
-        update_queue (:class:`queue.Queue`): Queue for the updates.
-        dispatcher (:class:`telegram.ext.Dispatcher`): Optional. Dispatcher that handles the
-            updates and dispatches them to the handlers.
-        running (:obj:`bool`): Indicates if the updater is running.
-        exception_event (:class:`threading.Event`): When an unhandled exception happens while
-            fetching updates, this event will be set. If :attr:`dispatcher` is not :obj:`None`, it
-            is the same object as :attr:`telegram.ext.Dispatcher.exception_event`.
+        update_queue (:class:`asyncio.Queue`): Queue for the updates.
 
-            .. versionadded:: 14.0
+    Args:
+        bot (:class:`telegram.Bot`): The bot used with this Updater.
+        update_queue (:class:`asyncio.Queue`): Queue for the updates.
 
     """
 
     __slots__ = (
-        'dispatcher',
-        'user_signal_handler',
         'bot',
-        'logger',
+        '_logger',
         'update_queue',
-        'exception_event',
         'last_update_id',
-        'running',
-        'is_idle',
-        'httpd',
+        '_running',
+        '_initialized',
+        '_httpd',
         '__lock',
-        '__threads',
+        '__polling_task',
     )
 
     def __init__(
-        self: 'Updater[BT, DT]',
-        *,
-        user_signal_handler: Callable[[int, object], Any] = None,
-        dispatcher: DT = None,
-        bot: BT = None,
-        update_queue: Queue = None,
-        exception_event: Event = None,
+        self,
+        bot: 'Bot',
+        update_queue: asyncio.Queue,
     ):
-        if not was_called_by(
-            inspect.currentframe(), Path(__file__).parent.resolve() / '_builders.py'
-        ):
-            warn(
-                '`Updater` instances should be built via the `UpdaterBuilder`.',
-                stacklevel=2,
-            )
-
-        self.user_signal_handler = user_signal_handler
-        self.dispatcher = dispatcher
-        if self.dispatcher:
-            self.bot = self.dispatcher.bot
-            self.update_queue = self.dispatcher.update_queue
-            self.exception_event = self.dispatcher.exception_event
-        else:
-            self.bot = bot
-            self.update_queue = update_queue
-            self.exception_event = exception_event
+        self.bot = bot
+        self.update_queue = update_queue
 
         self.last_update_id = 0
```

Comment:
is this intentionally kept public or? Because it's not documented to be public

Reply:
[It is](https://github.com/python-telegram-bot/python-telegram-bot/blob/41224509ced353f84731e800db14fd5fa6e41793/telegram/ext/_updater.py#L62), though? Anyway, it should be public - users might want to manually put updates into the queue

Reply:
Sorry, I meant the attribute last_update_id

Reply:
ah, right. then yes, that should probably be private 👌